File: BI-TJV/semestral/rest-client/app/src/main/java/cz/wrzecond/restsys/task/HttpResponse.java
package cz.wrzecond.restsys.task;

public final class HttpResponse {

    private final HttpStatus status;
    private final String locationHeader;
    private final String responseBody;

    public HttpResponse (HttpStatus status, String locationHeader, String responseBody) {
        this.status = status;
        this.locationHeader = locationHeader;
        this.responseBody = responseBody;
    }

    public HttpStatus getStatus () {
        return status;
    }

    public String getLocationHeader () {
        return locationHeader;
    }

    public String getResponseBody () {
        return responseBody;
    }

    @Override
    public String toString() {
        return "HttpResponse{" +
            "status=" + status +
            ", locationHeader='" + locationHeader + '\'' +
            ", responseBody='" + responseBody + '\'' +
            '}';
    }

}

File: BI-KOT/semestral/rest-client/app/src/main/java/cz/wrzecond/restsys/task/HttpResponse.kt
package cz.wrzecond.restsys.task

import androidx.appcompat.app.AppCompatActivity
import cz.wrzecond.restsys.R

data class HttpResponse (
    val status: HttpStatus,
    val locationHeader: String?,
    val responseBody: String?
) {
    fun getToastString (activity: AppCompatActivity)
    = if (status.isSuccess) activity.getString(R.string.delete_success)
      else activity.getString(R.string.delete_failure, status.name)
}
